src/main/java/appeng/recipes/factories/recipes/PartRecipeFactory.java

```diff
@@ -45,25 +60,149 @@ public IRecipe parse( JsonContext context, JsonObject json )
 		String type = JsonUtils.getString( json, "type" );
 		if( type.contains( "shaped" ) )
 		{
-			PartShapedCraftingFactory recipe = PartShapedCraftingFactory.factory( context, json );
-			CraftingHelper.ShapedPrimer primer = new CraftingHelper.ShapedPrimer();
-			primer.width = recipe.getWidth();
-			primer.height = recipe.getHeight();
-			primer.mirrored = JsonUtils.getBoolean( json, "mirrored", true );
-			primer.input = recipe.getIngredients();
-
-			return new PartShapedCraftingFactory( new ResourceLocation( AppEng.MOD_ID, "part_shaped_crafting" ), recipe.getRecipeOutput(), primer );
+			return shapedFactory( context, json );
 		}
 		else if( type.contains( "shapeless" ) )
 		{
-			PartShapelessCraftingFactory recipe = PartShapelessCraftingFactory.factory( context, json );
-
-			return new PartShapelessCraftingFactory( new ResourceLocation( AppEng.MOD_ID, "part_shapeless_crafting" ), recipe.getIngredients(), recipe
-					.getRecipeOutput() );
+			return shapelessFactory( context, json );
 		}
 		else
 		{
 			throw new JsonSyntaxException( "Applied Energistics 2 was given a custom recipe that it does not know how to handle!\n" + "Type should either be '" + AppEng.MOD_ID + ":shapeless' or '" + AppEng.MOD_ID + ":shaped', got '" + type + "'!" );
 		}
 	}
+
+	// Copied from ShapedOreRecipe.java, modified a bit.
+	private static ShapedOreRecipe shapedFactory( JsonContext context, JsonObject json )
+	{
+		String group = JsonUtils.getString( json, "group", AppEng.MOD_ID + ":part_shaped_crafting" );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the declaration of "group" closer to the code that uses it. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1941)